### PR TITLE
Feature/3 backup with velero

### LINKED ISSUE #3: Implement backup operation with velero as backup provider

### PR DESCRIPTION
**Pull request by jelemux**
Resolves #3

### CODE REVIEW

File: pkg/controllers/interfaces.go [hunk 1]
```diff
@@ -18,3 +18,15 @@ type eventRecorder interface {
 type controllerManager interface {
 	ctrl.Manager
 }
+
+// used for mocks
+
+//goland:noinspection GoUnusedType
+type etcdRegistry interface {
```
Comment (cesmarvin) [inline pkg/controllers/interfaces.go:25]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `etcdRegistry` is unused

File: pkg/controllers/interfaces.go [hunk 2]
```diff
@@ -18,3 +18,15 @@ type eventRecorder interface {
...
+	registry.Registry
+}
+
+//goland:noinspection GoUnusedType
+type etcdContext interface {
```
Comment (cesmarvin) [inline pkg/controllers/interfaces.go:30]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `etcdContext` is unused

File: pkg/maintenance/interfaces.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package maintenance
+
+import (
+	"github.com/cloudogu/cesapp-lib/registry"
+)
+
+type globalConfig interface {
```
Comment (cesmarvin) [inline pkg/maintenance/interfaces.go:7]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `globalConfig` is unused

File: pkg/provider/velero/backupProvider.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+func (p *provider) handleFailedBackup(backup *v1.Backup, err error) error {
+	p.recorder.Event(backup, corev1.EventTypeWarning, v1.ErrorOnCreateEventReason, err.Error())
+	return err
+}
+
+func waitForBackupCompletionOrFailure(veleroBackupChan <-chan watch.Event) error {
+eventLoop:
+	for veleroChange := range veleroBackupChan {
+		changedBackup := veleroChange.Object.(*velerov1.Backup)
```
Comment (cesmarvin) [inline pkg/provider/velero/backupProvider.go:98]: 🚫 **[checkstyle]** <forcetypeassert> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type assertion must be checked

File: pkg/provider/velero/interfaces.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+type veleroClientSet interface {
+	versioned.Interface
+}
+
+// The following interfaces are here to generate mocks.
+
+//goland:noinspection GoUnusedType
+type veleroInterface interface {
```
Comment (cesmarvin) [inline pkg/provider/velero/interfaces.go:27]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `veleroInterface` is unused

File: pkg/provider/velero/interfaces.go [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	velerov1.VeleroV1Interface
+}
+
+//goland:noinspection GoUnusedType
+type veleroBackupInterface interface {
```
Comment (cesmarvin) [inline pkg/provider/velero/interfaces.go:32]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `veleroBackupInterface` is unused

File: pkg/provider/velero/interfaces.go [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+	velerov1.BackupInterface
+}
+
+//goland:noinspection GoUnusedType
+type veleroBackupStorageLocationInterface interface {
```
Comment (cesmarvin) [inline pkg/provider/velero/interfaces.go:37]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `veleroBackupStorageLocationInterface` is unused

File: pkg/provider/velero/interfaces.go [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+	velerov1.BackupStorageLocationInterface
+}
+
+//goland:noinspection GoUnusedType
+type veleroDeleteBackupRequest interface {
```
Comment (cesmarvin) [inline pkg/provider/velero/interfaces.go:42]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `veleroDeleteBackupRequest` is unused

File: pkg/provider/velero/interfaces.go [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+	velerov1.DeleteBackupRequestInterface
+}
+
+//goland:noinspection GoUnusedType
+type ecosystemWatch interface {
```
Comment (cesmarvin) [inline pkg/provider/velero/interfaces.go:47]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `ecosystemWatch` is unused

File: pkg/controllers/restoreReconciler_test.go [hunk 10]
```diff
@@ -27,7 +31,7 @@ func Test_restoreReconciler_Reconcile(t *testing.T) {
 		sut := &restoreReconciler{}
 
 		// when
-		actual, err := sut.Reconcile(testCtx, request)
+		actual, err := sut.Reconcile(context.TODO(), request)
```
Comment (jelemux) [inline pkg/controllers/restoreReconciler_test.go:34]: Why not simply use `testCtx`?

File: CHANGELOG.md [hunk 11]
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
```
Comment (jelemux) [inline CHANGELOG.md:7]: We need a changelog entry

**Pull request merged**